HVZ.Persistence.MongoDB/Repos/GameRepo.cs

```diff
@@ -225,6 +232,109 @@ public async Task<Game> LogTag(string gameId, string taggerUserId, string tagRec
         return newGame;
     }
 
+    public async Task<Game> AddPlayerToOzPool(string gameId, string playerId)
+    {
+        Game game = await GetGameById(gameId);
+        Player? player = await FindPlayerByUserId(gameId, playerId);
+        if (player is null)
+        {
+            throw new ArgumentException($"Could not find player with GameId {playerId} in Game {gameId}");
+        }
+        if (game.OzPool.Contains(playerId))
+        {
+            throw new ArgumentException($"Player with UserId {playerId} is already in OZ Pool for game {gameId}");
+        }
+
+        game.OzPool.Add(playerId);
+        Game newGame = await Collection.FindOneAndUpdateAsync<Game>(g => g.Id == gameId,
+            Builders<Game>.Update.Set(g => g.OzPool, game.OzPool),
+            new FindOneAndUpdateOptions<Game, Game>() { ReturnDocument = ReturnDocument.After }
+        );
+        OnJoinOzPool(new(newGame, playerId));
+        _logger.LogTrace($"Player {playerId} has been added to the OZ pool in Game {gameId}");
+        return newGame;
+    }
+
+    public async Task<Game> RemovePlayerFromOzPool(string gameId, string playerId)
+    {
+        Game game = await GetGameById(gameId);
+        Player? player = await FindPlayerByUserId(gameId, playerId);
+        if (player is null)
+        {
+            throw new ArgumentException($"Could not find player with UserId {playerId} in Game {gameId}");
+        }
+        if (!game.OzPool.Contains(playerId))
+        {
+            throw new ArgumentException($"Player with GameId {playerId} is not in the OZ pool for Game {gameId}");
+        }
+
+        game.OzPool.Remove(playerId);
+        Game newGame = await Collection.FindOneAndUpdateAsync<Game>(g => g.Id == gameId,
+            Builders<Game>.Update.Set(g => g.OzPool, game.OzPool),
+            new FindOneAndUpdateOptions<Game, Game>() { ReturnDocument = ReturnDocument.After }
+        );
+        OnLeaveOzPool(new(newGame, playerId));
+        _logger.LogTrace($"Player {playerId} has been removed from the OZ pool in Game {gameId}");
+        return newGame;
+    }
+
+    public async Task<Game> RandomOzs(string gameId, int count, string instigatorId)
```

Comment:
```suggestion
    public async Task<Game> AssignRandomOzsFromPool(string gameId, int count, string instigatorId)
```
more clear what the method is doing